challenge_6/python/ning/challenge_6.py

```diff
@@ -0,0 +1,19 @@
+def ranges(input_list):
+    output_elements = list(input_list)  # copy, not assign
+    output_list = []
+
+    for integer in input_list:
+        if (integer + 1 not in input_list and
```

Comment:
Because you are using a `list` and not a `set` this `integer not in input_list` runs in O(n) time, making your algorithm run in O(n^2).  You should be able to solve this algorithm in O(n) time.